
# The Technology and Statistics behind COVID-19 Testing
##### A BENG183 Final Paper written by Aaron Oliver, Nikhill Joshi, and Sara Safa.

 1. [Introduction](#1)
 2. [Testing Basics](#2)
 3. [The Molecular Test](#3) <br>
	 3.1. [The First Steps of Development](#31) <br>
	 3.2. [The Mechanism of SARS-CoV-2 Detection](#32) <br>
	 3.3. [Sample Pooling: More Results with Less Tests](#33) 
 4. [Conclusion](#4)
 5. [References](#5)
 
## 1. Introduction <a name="1"></a>
Widespread infection by the severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), the cause of coronavirus disease 2019 (COVID-19), has shaped the means by which we learn, work, and interact. In the absence of any known cure for this disease, a critical line of defense against the pandemic has been frequent and fast testing. Despite FDA approval of a vaccine just this week [[15]](#c15), we are still months away (in the most optimistic of projections) from herd immunity and the beginning of an end to the pandemic that has ravaged the globe more than any other event in our lifetimes. Thus, a strong and unwavering testing initiative will continue to save many lives in the upcoming year. The following chapter will explore the mechanisms and statistical concepts that power COVID-19 testing.

## 2. Testing Basics <a name="2"></a>
The three most common tests used to detect COVID-19 infection are the molecular, antigen, and antibody tests. The table below, adapted from a document released by the Federal Drug Administration [[1]](#c1), shows the three most common types of COVID-19 testing currently in use in the United States. Each test fulfills a niche in the battle against COVID-19, and all three protocols are used extensively. The molecular test and antigen test use viral RNA and viral protein to diagnose active infection, respectively. Meanwhile, the antibody test uses a patient's blood sample to detect remnants of past COVID-19 infection.
|Test | Molecular Test | Antigen Test | Antibody Test|
|:----:|:-:|:-:|:-:|
|Aims to find... | Active COVID-19 infection | Active COVID-19 infection | Past COVID-19 infection |
|Using... | Reverse transcription qPCR to amplify sample RNA | Antigen binding to detect viral proteins | Detection of antibodies in blood sample|
|Results in...|Days|Hours|Days|
|Sensitivity | Less false negatives | More false negatives | Less false negatives |

As with any traditional diagnosis test, we can create a confusion matrix that describes the different outcome for a COVID-19 test. To extend the statistical concepts discussed in BENG 183 into real world consequences, consider the effects of a False Negative for a COVID-19 test compared to a False Positive. A patient that recieves a False Positive diagnosis will isolate unnecessarily in their home even though they can not spread COVID-19. Although this is inconvenient, compare this to the consequences of a False Negative result. In this scenario, a carrier of SARS-CoV-2 may not isolate in their homes, and thus they have the potential to spread COVID-19 further around their community. 

|COVID-19 Confusion Matrix |Patient has COVID-19 | Patient does **not** have COVID-19 |
|:-:|:-:|:-:|
|Test detects SARS-CoV-2 | True Positive | False Positive |
|Test fails to detect SARS-CoV-2 | False Negative | True Negative |

Recall that specificity is defined as the number of True Positives divided by the sum of True Positives and False Negatives. Essentially, a high specificity stems from minimizing the number of False Negatives. Operating under the assumption that our main goal is to minimize the spread of COVID-19, the most important feature of a COVID-19 test is to maximize sensitivity above all else. An alternate test that sacrifices sensitivity for fewer False Positives will reduce inconvenience for patients that would have ended up receiving False Positive diagnoses, but this test would ultimately be **less effective** in preventing the spread of COVID-19.

## 3. The Molecular Test <a name="3"></a>

The remainder of this chapter will focus on the molecular test due to its improved specificity and relationship to other topics in the BENG 183 curriculum.

### 3.1 The First Steps of Development <a name="31"></a>
The first complete sequence of the SARS-CoV-2 viral genome was generated by Fan Wu *et al* in January 2020, with their findings published in *Nature* in early February [[2]](#c2). The complete genome of the virus was determined using a reverse-transcription polymerase chain reaction (RT-PCR), in which reverse transcriptase converts the viral RNA to completementary DNA (cDNA) that could then be amplified using PCR. Exploration into viral phylogeny showed that SARS-CoV-2 is related to the SARS virus that caused the global SARS pandemic in 2002-2003 [[2]](#c2). The original sequence remains accessible on Genbank under the identifier MN908947.1 [[14]](#c14), although newer, more refined viral genomes are also available. The image below shows the viral genome of SARS-CoV-2 [[6]](#c6), where colored blocks between the 5' and 3' UTR represent different genes and intervals below the genome show the binding locations of SARS-CoV-2-specific PCR primers.

<img src="images/covid_genome.PNG" width="700">

Once the SARS-CoV-2 genome was publicly available, the next step was to create primers for use in PCR to amplify SARS-CoV-2 sequence. PCR primers that specifically target SARS-CoV-2 genes enable its detection with high confidence using only a small amount of input nucleic acid. Primer design is an iterative process where a gene of interest is identified, and then primers that bind to this gene are repeatedly optimized at both the computational (*in silico*) and biochemical (*in vitro*) level. While primers based on other SARS viruses were available even prior to the publication of a SARS-CoV-2 sequence, COVID-19 specific primers were still undergoing iterations of refinement in June 2020 [[6]](#c6).

### 3.2 The Mechanism of SARS-CoV-2 Detection <a name="32"></a>
Once primers of acceptable quality are finalized, the next step is to process single samples using an RT-PCR test that analyzes data in real time (RT-qPCR). This real-time quantitative PCR process detects and quantifies the replication of genetic material interest through fluorescence. The starting input of RNA is first transcribed into complementary DNA using reverse transcriptase. The PCR primers, developed to bind specifically to SARS-CoV-2 genes, will attach to the viral cDNA at each replication cycle of PCR. The fluorescence emission at each cycle is measured and plotted over time to determine if the fluorescence levels exceed a predetermined cutoff value, signifying the presence of SARS-CoV-2.

The fluorescence mechanism is shown in the image below [[8]](#c8). First, a specialized TaqMan probe anneals to a region of a SARS-CoV-2 gene that is amplified during a PCR round. As the polymerization enzyme approaches the probe, Taq polymerase degrades the probe and causes fluorescence as the probe cleaves. After the probe degrades and emits fluorescence, it also breaks bonds with the template DNA so that the polymerization step can conclude. At the end of one round RT-qPCR, the probe is destroyed and the standard PCR products are formed.

<img src="images/Taqman.png" width="400">

RT-qPCR is used for molecular COVID-19 testing rather than the more popular RNA-seq technology for a few reasons. Although RNA-seq can identify novel sequences and alternate viral variants, the lower sensitivity and larger data output make it difficult to quickly detect viral sequences. Meanwhile, RT-qPCR amplifies genetic sequence so that it has a higher sensitivity for detecting weakly expressed genes [[7]](#c7). Even though RT-qPCR can only identify known sequences through targeted fluorescence, this is not a major issue because the molecular test only needs to target specific COVID-19 genes.
 
### 3.3 Sample Pooling: More Results with Less Tests <a name="33"></a>
To ramp up testing scalability, an individual testing protocol transforms into a pooling strategy that processes more tests in less time. The image below [[10]](#c10) shows a hypothetical COVID-19 testing protocol that is more efficient than a naive approach of one test for each sample. In the United States, RT-qPCR pooling for COVID-19 was not approved by the Federal Drug Administration until July 2020 [[5]](#c5), months after widespread testing initiatives for SARS-CoV-2.

<img src="images/pooling_method.jpg" width="400">

First, the samples (96 in this example) are placed into an 8 × 12 matrix. Then, a subsample from each of the samples in a row are combined into a mixture sample representative for that row. This mixture sample then undergoes the RT-qPCR test to detect COVID-19. Repeat this process for all 8 of the rows, as well as for all 12 of the columns. It is possible to perform multiple RT-qPCR tests using RNA from a sample multiple times because the PCR replication step is sensitive enough to generate detectable fluorescence from a very small amount of input nucleic acid.

In the next round of testing, we perform individual testing on any sample that is in both a COVID-19 positive row and column. In scenarios where the percentage of SARS-CoV-2 containing samples is relatively low, the pruning provided by the row-wise and column-wise testing greatly reduces the number of individual tests that we need to perform.

Through simulation, previous research by Baha Abdalhamid *et al* in June 2020 [[11]](#c11) discovered that the expected number of tests required to diagnose patients using a pooling strategy is more than 50% less than using individual testing alone. The graph below from the Abdalhamid *et al* study shows that this efficiency gain holds for various pooling sizes and test specificities. The apex of the graph at a group size of 5 suggests that a 5 x 5 matrix which tests 25 samples under the protocol shown above would be the most efficient configuration tested in this study.

<img src="images/pooling_graph.jpg" width="700">

The Abdalhamid *et al* study, among other literature on sample pooling, rely on an important assumption about the samples - the vast majority of samples should **not** contain SARS-CoV-2. As a thought experiment, we can consider three scenarios in which we can test a matrix of *n* by *m* samples.

In Scenario 1, we use sample pooling on a batch of samples where all of the samples test negative for COVID-19. To get this result, the total number of tests we needed to perform was *n* for each of the rows and *m* for each of the columns, for a total number of tests of *n + m*.

In Scenario 2, we use individual testing and do not make any assumptions about the prevalence of COVID-19 in our sampled population. The total number of test required will be *n* × *m*, since we perform a single test on each sample in our array.

A worst case would be Scenario 3, where we use sample pooling on a batch of samples where every sample tests positive for COVID-19. Initially, we would have to perform the row-wise and column-wise testing for *n + m* tests. Then, since we observed that every row and every column contains SARS-CoV-2, we would have to perform individual testing on every sample (*n* × *m* tests), for a total of *(n* × *m) + n + m* tests, much worse than if we just used individual testing from the start!

A common threshold, used by Abdalhamid *et al*, is to proceed with pooling if we can assume that less than 10% of a batch of incoming samples contains SARS-CoV-2. This means that test pooling can be effective in states like Hawai'i, which has a 2.0% positive test rate at the time of this paper (December 14, 2020) [[12]](#c12). However, pooling would likely be detrimental to testing efficiency if implemented in a state like South Dakota, with widespread COVID-19 infection and a positive test rate of 15.1% [[12]](#c12). For reference, the positive test rate of San Diego County is 8.2% [[12]](#c12), while the positive test rates for students under the Return to Learn Program at UCSD is 0.8% [[13]](#c13).

## 4. Conclusion <a name="4"></a>
In this chapter, we explored the development, mechanism, and math behind the COVID-19 molecular test. Guided by a publicly available viral genome, thousands of researchers have collaborated and contributed to new breakthroughs about SARS-CoV-2 at speeds unlike anything in recent scientific history. The story of COVID-19 is still being written, and it is possible that new publications will revolutionize how we test for this disease in the coming months. We hope that this chapter serves as a useful snapshot about how COVID-19 tests were developed and used in the midst of this pandemic.

Although this chapter focused on the molecular test, all three of the testing strategies discussed in this chapter have their role in combating the spread of COVID-19. Antigen tests are fast enough to act as near real-time screens for workers and consumers of essential services. Molecular tests can detect SARS-CoV-2 in patients with high specificity and high scalability. Antibody tests generate useful data about past infections to help public health officials predict where and when the next outbreaks will occur. It is only through the combination of these tools that our society can continue to stand strong during a public health crisis unlike anything we have had to endure. We hope that future readers can soon look back on the COVID-19 pandemic as a historical anomaly, enjoying the full protection of a vaccine that will hopefully be completely distributed by the next time BENG 183 is taught.

## 5. References <a name="5"></a>
###### Some references listed here are not used in this paper but are referenced in the associated presentation, and vice versa.
1.  Federal Drug Administration (FDA), [https://www.fda.gov/consumers/consumer-updates/coronavirus-disease-2019-testing-basics](https://www.fda.gov/consumers/consumer-updates/coronavirus-disease-2019-testing-basics). <a name="c1"></a>
    
2.  Wu, F., Zhao, S., Yu, B. et al. A new coronavirus associated with human respiratory disease in China. Nature, 2020. [https://doi.org/10.1038/s41586-020-2008-3](https://doi.org/10.1038/s41586-020-2008-3). <a name="c2"></a>
    
3.  Corman, Victor M et al. Detection of 2019 novel coronavirus (2019-nCoV) by real-time RT-PCR. European Communicable Disease Bulletin, Volume 25, Issue 3, 2020. doi:10.2807/1560-7917.ES.2020.25.3.2000045. <a name="c3"></a>
    
4.  Jasper Fuk-Woo Chan, et al. Improved Molecular Diagnosis of COVID-19 by the Novel, Highly Sensitive and Specific COVID-19-RdRp/Hel Real-Time Reverse Transcription-PCR Assay Validated In Vitro and with Clinical Specimens. Journal of Clinical Microbiology, Apr 2020. DOI: 10.1128/JCM.00310-20. <a name="c4"></a>
    
5.  Federal Drug Administation, [https://www.fda.gov/news-events/press-announcements/coronavirus-covid-19-update-fda-issues-first-emergency-authorization-sample-pooling-diagnostic](https://www.fda.gov/news-events/press-announcements/coronavirus-covid-19-update-fda-issues-first-emergency-authorization-sample-pooling-diagnostic). <a name="c5"></a>
    
6.  Park, M., Won, J., Choi, B.Y. et al. Optimization of primer sets and detection protocols for SARS-CoV-2 of coronavirus disease 2019 (COVID-19) using PCR and real-time PCR. Exp Mol Med Volume 52, 2020, Pages 963–977, [https://doi.org/10.1038/s12276-020-0452-7](https://doi.org/10.1038/s12276-020-0452-7). <a name="c6"></a>
    
7.  Derveaux, Stefaan et al. “How to do successful gene expression analysis using real-time PCR.” Methods, Volume 50, Issue 4, 2010, doi:10.1016/j.ymeth.2009.11.001. <a name="c7"></a>
    
8.  Image from: Wikipedia, [https://en.wikipedia.org/wiki/Reverse_transcription_polymerase_chain_reaction](https://en.wikipedia.org/wiki/Reverse_transcription_polymerase_chain_reaction). <a name="c8"></a>
    
9.  IGI Testing Consortium., Amen, A.M., Barry, K.W. et al. Blueprint for a pop-up SARS-CoV-2 testing lab. Nat Biotechnol Volume 38, 2020, Pages 791–797, [https://doi.org/10.1038/s41587-020-0583-3](https://doi.org/10.1038/s41587-020-0583-3). <a name="c9"></a>
    
10.  Image from: Nasa Sinnott-Armstrong, Daniel L. Klein, Brendan Hickey. Evaluation of Group Testing for SARS-CoV-2 RNA, medRxiv, March 2020. [https://doi.org/10.1101/2020.03.27.20043968](https://doi.org/10.1101/2020.03.27.20043968). <a name="c10"></a>
    
11.  Baha Abdalhamid, Christopher R Bilder, Emily L McCutchen, Steven H Hinrichs, Scott A Koepsell, Peter C Iwen, Assessment of Specimen Pooling to Conserve SARS CoV-2 Testing Resources, American Journal of Clinical Pathology, Volume 153, Issue 6, June 2020, Pages 715–718, [https://doi.org/10.1093/ajcp/aqaa064](https://doi.org/10.1093/ajcp/aqaa064). <a name="c11"></a>
    
12.  Data aggregation by Covid Act Now,  [https://covidactnow.org/](https://covidactnow.org/). <a name="c12"></a>

13. UC San Diego Return to Learn dashboard, [https://returntolearn.ucsd.edu/dashboard/index.html](https://returntolearn.ucsd.edu/dashboard/index.html). <a name="c13"></a>

14. Genbank, Wuhan seafood market pneumonia virus isolate Wuhan-Hu-1, complete genome. [https://www.ncbi.nlm.nih.gov/nuccore/MN908947.1](https://www.ncbi.nlm.nih.gov/nuccore/MN908947.1).  <a name="c14"></a>

15. Federal Drug Administration, [https://www.fda.gov/news-events/press-announcements/fda-takes-key-action-fight-against-covid-19-issuing-emergency-use-authorization-first-covid-19](https://www.fda.gov/news-events/press-announcements/fda-takes-key-action-fight-against-covid-19-issuing-emergency-use-authorization-first-covid-19).  <a name="c15"></a>
